src/api/expense.controller.js

```diff
@@ -0,0 +1,139 @@
+const { expenseService } = require('../services/expense.service.js');
+const { userService } = require('../services/user.service.js');
+
+const isValidDate = (dateString) => {
+  const date = new Date(dateString);
+
+  return !isNaN(date.getTime());
+};
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const categoryList = categories ? categories.split(',') : [];
+
+  if (from && !isValidDate(from)) {
+    return res.status(400).json({ message: "'from' is not a valid date." });
+  }
+
+  if (to && !isValidDate(to)) {
+    return res.status(400).json({ message: "'to' is not a valid date." });
+  }
+
+  const expenses = await expenseService.getExpenses({
+    userId,
+    categoryList,
+    from,
+    to,
+  });
+
+  res.json(expenses);
+};
+
+const createExpense = async (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount || amount <= 0) {
+    return res.status(400).json({
+      message: 'userId, spentAt, title, and a positive amount are required.',
+    });
+  }
```

Comment:
The `createExpense` function should include a validation check for the `spentAt` field to ensure it is a valid date. This is important to maintain consistency with the `updateExpense` function, which already performs this check.